references/validationFunctions/common/ajccValidation.js

```diff
@@ -1,37 +1,56 @@
-            const stagingName = $name.trim().toLowerCase().split('_tumour_staging_system')[0];
-            /* Perform validation only if an AJCC value was selected */
-            if (/^(AJCC)\b/i.test($field)){
-                const requiredFields = [
-                    `${stagingName}_m_category`,
-                    `${stagingName}_n_category`,
-                    `${stagingName}_t_category`
-                ];
-                const convertedRow = Object.fromEntries(Object.entries($row).map(([fieldName,fieldVal]) => [fieldName.toLowerCase(), fieldVal]));
-                /* Check for contigous spaces wrapped with quotes (empty strings) */
-                const checkforEmpty = (entry) => {return /^\s+$/g.test(decodeURI(entry).replace(/^"(.*)"$/, '$1'))};
-                /* search for fields with falsy values, given the field exists */
-                const emptyFields = requiredFields.filter(field => Object.keys(convertedRow).includes(field) && (!convertedRow[field] || checkforEmpty(convertedRow[field])))
-                if (emptyFields.length){
-                    result = {valid: false, "message": `The following fields are required when ${$name} is set to an AJCC option: ${emptyFields}`};
-                }
-            }
-        }
-        return result;
-    })();
+    /* This is not a required field, so first ensure that it exists */
+    if ($field) {
+      /* Contingent on the naming system for tumour staging systems to remain consistent */
+      const stagingName = $name
+        .trim()
+        .toLowerCase()
+        .split('_tumour_staging_system')[0];
+      const requiredFields = [
+        `${stagingName}_m_category`,
+        `${stagingName}_n_category`,
+        `${stagingName}_t_category`,
+      ];
+      const convertedRow = Object.fromEntries(
+        Object.entries($row).map(([fieldName, fieldVal]) => [fieldName.toLowerCase(), fieldVal]),
+      );
+      /* Check for contigous spaces wrapped with quotes (empty strings) */
+      const checkforEmpty = entry => {
+        return /^\s+$/g.test(decodeURI(entry).replace(/^"(.*)"$/, '$1'));
+      };
+      /* search for fields with falsy values, given the field exists */
+      const emptyFields = requiredFields.filter(
+        field =>
+          Object.keys(convertedRow).includes(field) &&
+          (!convertedRow[field] || checkforEmpty(convertedRow[field])),
+      );
+
+      /* The fields should be provided IF and ONLY IF the AJCC regex passes */
+      if (/^(AJCC)\b/i.test($field) && emptyFields.length) {
+        result = {
+          valid: false,
+          message: `The following fields are required when ${$name} is set to an AJCC option: ${emptyFields}`,
+        };
+      } else if (!/^(AJCC)\b/i.test($field) && emptyFields.length != requiredFields.length) {
+        result = {
+          valid: false,
+          message: `The following fields cannot be provided when ${$name} is not set to an AJCC option: ${requiredFields.filter(
```

Comment:
Do we want to format this message in the same way as other messages, with a list starting with dashes? right now it looks like this shows an array in a square bracket...

Just want to be consistent.

Reply:
Yeah that's a good point! I'll do the list with dashes for this message.